applications/ConstitutiveLawsApplication/custom_utilities/autom_differentiation_tangent_utilities.cpp

```diff
@@ -0,0 +1,3528 @@
+  const double cr_Ct149 = 0.25*cr_Ct40*cr_Ct99;
+  const double cr_Ct150 = -cr_Ct149*cr_Ct43;
+  const double cr_Ct151 = -cr_Ct149*cr_Ct38;
+  const double cr_Ct152 = cr_Ct94 - 2.0;
+  const double cr_Ct153 = cr_Ct150 + cr_Ct151 + cr_Ct152*cr_Ct33*cr_Ct68;
+  const double cr_Ct154 = cr_Ct53*cr_Ct84;
+  const double cr_Ct155 = cr_Ct154*cr_Ct18;
+  const double cr_Ct156 = 0.22222222222222227*cr_Ct155;
+  const double cr_Ct157 = cr_Ct112*cr_Ct135*cr_Ct52*(cr_Ct136*cr_Ct81 - cr_Ct137*cr_Ct77);
+  const double cr_Ct158 = cr_Ct134*cr_Ct157 + cr_Ct148 + cr_Ct153*cr_Ct156;
+  const double cr_Ct159 = 1.0/(Gf*Young*std::pow(cr_Ct9, 2)/(characteristic_length*std::pow(threshold_compression, 2)) - 0.5);
+  const double cr_Ct160 = cr_Ct159;
+  const double cr_Ct161 = cr_Ct160*cr_Ct18*cr_Ct87;
+  const double cr_Ct162 = cr_Ct143*cr_Ct161;
+  const double cr_Ct163 = std::exp(cr_Ct160*(-2.0*cr_Ct10*cr_Ct86/(cr_Ct88*threshold_compression) + 1.0));
+  const double cr_Ct164 = cr_Ct163*cr_Ct18;
+  const double cr_Ct165 = cr_Ct164*cr_Ct8;
+  const double cr_Ct166 = cr_Ct91*nu;
+  const double cr_Ct167 = cr_Ct98*cr_Ct99;
+  const double cr_Ct168 = cr_Ct101*cr_Ct95 + cr_Ct104 + cr_Ct167;
+  const double cr_Ct169 = cr_Ct119*cr_Ct21;
+  const double cr_Ct170 = 2.598076211353316*cr_Ct126;
+  const double cr_Ct171 = cr_Ct121*cr_Ct127;
+  const double cr_Ct172 = 2.598076211353316*cr_Ct125;
+  const double cr_Ct173 = cr_Ct130*cr_Ct21;
+  const double cr_Ct174 = cr_Ct128*(cr_Ct123 + cr_Ct170*cr_Ct47 + cr_Ct171 - cr_Ct172*(cr_Ct121*cr_Ct65 + cr_Ct126*cr_Ct61)) + cr_Ct132*cr_Ct168 + cr_Ct168*cr_Ct173;
+  const double cr_Ct175 = 0.5*cr_Ct139;
+  const double cr_Ct176 = cr_Ct168*cr_Ct169 + cr_Ct174*cr_Ct175 + cr_Ct93;
+  const double cr_Ct177 = cr_Ct57*cr_Ct8;
+  const double cr_Ct178 = cr_Ct152*cr_Ct177;
+  const double cr_Ct179 = cr_Ct177*cr_Ct33*cr_Ct99;
+  const double cr_Ct180 = cr_Ct150 - cr_Ct178*cr_Ct38 + cr_Ct179;
+  const double cr_Ct181 = 0.5*cr_Ct157;
+  const double cr_Ct182 = cr_Ct148 + cr_Ct156*cr_Ct180 + cr_Ct174*cr_Ct181;
+  const double cr_Ct183 = cr_Ct102 + cr_Ct103*cr_Ct95 + cr_Ct167;
+  const double cr_Ct184 = cr_Ct128*(cr_Ct124 + cr_Ct170*cr_Ct45 + cr_Ct171 - cr_Ct172*(cr_Ct121*cr_Ct61 + cr_Ct126*cr_Ct65)) + cr_Ct132*cr_Ct183 + cr_Ct173*cr_Ct183;
+  const double cr_Ct185 = cr_Ct169*cr_Ct183 + cr_Ct175*cr_Ct184 + cr_Ct93;
+  const double cr_Ct186 = cr_Ct151 - cr_Ct178*cr_Ct43 + cr_Ct179;
+  const double cr_Ct187 = cr_Ct148 + cr_Ct156*cr_Ct186 + cr_Ct181*cr_Ct184;
+  const double cr_Ct188 = cr_Ct18*r_strain[3];
+  const double cr_Ct189 = 2.598076211353316*r_strain[5];
+  const double cr_Ct190 = 5.196152422706632*cr_Ct57;
+  const double cr_Ct191 = 8.0*cr_Ct129;
+  const double cr_Ct192 = cr_Ct131*cr_Ct74;
+  const double cr_Ct193 = cr_Ct114*(-cr_Ct189*r_strain[4] + cr_Ct190*cr_Ct62) + cr_Ct191*r_strain[3] + cr_Ct192*r_strain[3];
+  const double cr_Ct194 = cr_Ct118*cr_Ct188 + cr_Ct138*cr_Ct193;
+  const double cr_Ct195 = cr_Ct144;
```

Comment:
Sorry for butting in unrequested. I have no idea what's being computed here, but is there any chance of implementing it in a way that is maintainable? This is practically impossible to debug / extend / adapt?

Reply:
The point is to precisely implement it in that way. This code is "always" correct as it is generated from the automatic differentiation meaning that, if there is any error, it is necessarily located in the python scripts within the `symbolic_generation` folder, which are "super" easy to be debugged.

Reply:
ah I see, so this is auto-generated? @AlejandroCornejo can you please mention that in a comment, and maybe link the script with its arguments, so that anyone can reproduce it?

Reply:
ok, Actually the python files that lead to this c++ code are also added in this PR. I'll comment this in the .h file